k8s-pb-codegen/src/main.rs

```diff
@@ -50,18 +54,26 @@ fn main() -> Result<()> {
     // NB: FDS fields: https://github.com/tokio-rs/prost/blob/32bc87cd0b7301f6af1a338e9afd7717d0f42ca9/prost-types/src/protobuf.rs#L1-L7
     // FDS usage: https://github.com/tokio-rs/prost/blob/32bc87cd0b7301f6af1a338e9afd7717d0f42ca9/prost-build/src/lib.rs#L765-L825
     for f in fds.file {
-        if let Some(pkg) = f.package {
-            for m in f.message_type {
-                if let Some(name) = m.name {
-                    let path = format!("{}.{}", pkg, name);
-                    if let Some(resource) = resources.get(&path) {
-                        info!("generating resource {}", path);
-                        debug!("{:?}", resource);
-                    }
-                }
+        if let Some(package_name) = f.package {
+            let mut pkg_rs = OpenOptions::new()
+                .append(true)
+                .open(tmp_dir.join(format!("{}.rs", package_name)))
+                .unwrap();
+
+            for msg in f.message_type {
+                let message_name = match msg.name {
+                    Some(ref name) => name,
+                    None => continue,
+                };
+                let path = format!("{}.{}", package_name, message_name);
+                let resource = match resources.get(&path) {
+                    Some(resource) => resource,
+                    None => continue,
+                };
+                append_trait_impl(&mut pkg_rs, message_name, resource);
```

Comment:
the `continue` here goes to the innermost loop, so this is equivalent to:
```suggestion
                if let Some(resource) = resources.get(&path) {
                    append_trait_impl(&mut pkg_rs, message_name, resource);
                }
```